packages/bot-config-utils/src/bot-config-utils.ts

```diff
@@ -28,33 +28,113 @@ type Conclusion =
+ */
+interface File {
+  content: string | undefined;
+}
+
+/**
+ * This function is used for comforting typescript's type system to
+ * deal with the response from `octokit.repos.getContent`.
+ */
+function isFile(file: File | unknown): file is File {
+  return (file as File).content !== undefined;
 }
 
-export interface ValidateConfigResult {
+/**
+ * The return value of `validateConfig` function.
+ *
+ * @template T
+ * @prop {boolean} isValid - The result of the schema validation.
+ * @prop {string} errorText - The error message from the validation.
+ * @prop {T} config - The loaded config object,only available when the
+ *   validation succeeds.
+ */
+export interface ValidateConfigResult<T> {
   isValid: boolean;
   errorText: string | undefined;
+  config?: T;
 }
 
-interface File {
-  content: string | undefined;
+/**
+ * The optionnal arguments for `validateConfig`.
+ *
+ * @prop {Array<object>} additionalSchemas - Additional schemas for Ajv schema
+ *   validator. When you have complex schema that used definition in different
+ *   files, you need to give all the schema definitions to Ajv.
+ */
+export interface ValidateConfigOptions {
+  additionalSchemas?: Array<object>;
 }
 
-function isFile(file: File | unknown): file is File {
-  return (file as File).content !== undefined;
+/**
+ * It loads the given string as yaml, then validates against the given schema.
+ *
+ * @template T
+ * @param {string} configYaml - The string representation of the config.
+ * @param {object} schema - The schema definition.
+ * @param {ValidateConfigOptions} options - Optional arguments for validation.
+ *
+ * @return {ValidateConfigResult<T>}
+ */
+export function validateConfig<T>(
+  configYaml: string,
+  schema: object,
+  options: ValidateConfigOptions
+): ValidateConfigResult<T> {
+  const ajv = new Ajv();
+  if (options.additionalSchemas) {
+    for (const schema of options.additionalSchemas) {
+      ajv.addSchema(schema);
+    }
+  }
+  const validateSchema = ajv.compile(schema);
+  let isValid = false;
+  let errorText: string | undefined;
+  let config: T | undefined;
+  try {
+    const candidate = yaml.load(configYaml) as unknown as T;
+    isValid = validateSchema(candidate);
+    if (isValid) {
+      config = candidate;
+    } else {
+      errorText = JSON.stringify(validateSchema.errors, null, 4);
+    }
+  } catch (err) {
+    // failed to load the yaml file
+    errorText = 'the given config is not valid YAML 😱 \n' + err.message;
+  }
+  return {isValid: isValid, errorText: errorText, config: config};
 }
 
+/**
+ * A class for validating the config changes on pull requests.
+ *
+ * @template T
+ */
 export class ConfigChecker<T> {
-  private ajv: Ajv;
   private schema: object;
+  private additionalSchemas: Array<object>;
   private configPath: string;
   private badConfigPaths: Array<string>;
   private configName: string;
   private config: T | null;
-  constructor(schema: object, configFileName: string) {
+
+  /**
```

Comment:
This comment is unnecessary.  Please remove it.

Reply:
ok

Reply:
Removed